src/libs/antares/inifile/inifile.cpp

```diff
@@ -237,29 +238,20 @@ uint IniFile::Section::size() const
 
 bool IniFile::open(const AnyString& filename, bool warnings)
 {
-    // clear
     clear();
-    pFilename = filename;
+    std::string filePath = filename.to<std::string>();
+    const unsigned int lineMaxSize = 2048;
 
-    // Load the file
-    IO::File::Stream file;
-    if (file.open(filename))
+    std::ifstream file(filePath);
+    if (file.is_open())
```

Comment:
Proposal
```cxx
IniFile::readStream(std::istream& in, bool warnings)
{
   // most generic implementation
}
IniFile::open(const std::filesystem::path& file, bool warnings) 
{
   std::ifstream f(path);
   if (f.is_open())
   {
      return readStream(f, warnings);
   }
}
```
So we can have easy in-memory tests with `std::istringstream`

See [this diagram](https://en.cppreference.com/w/cpp/io) for more details.

Reply:
Reminder : as @sylvlecl said, **readStream(...)** should be made **public** in order to test it, and because it could be useful to have it **public** some day.

Reply:
Done